src/amber/router/request.cr

```diff
@@ -0,0 +1,20 @@
+require "./**"
+
+class HTTP::Request
+  # This enables optional cookies store processing
+  @cookies_store : Amber::Router::Cookies::Store?
+
+  # Only enable advance cookies store if the Amber::Server.cookie_enabled?
```

Comment:
Actually I was thinking of moving everything out the context and move it to the request. Cookies live in the request not in the context such as the params and other information. I decided not to move it now because it would have break a lot of things. 

Also by moving it to the request means that we don't have to worry about the cookies until we actually use in a pipeline and/or controller